container/container.go

```diff
@@ -429,14 +429,19 @@ func (container *Container) ShouldRestart() bool {
 
 // AddMountPointWithVolume adds a new mount point configured with a volume to the container.
 func (container *Container) AddMountPointWithVolume(destination string, vol volume.Volume, rw bool) {
+	plat := container.Platform
+	if plat == "" {
+		plat = runtime.GOOS
+	}
```

Comment:
Do you think it would make sense to move this default into `NewParser()` ?

I guess a lot of code is calling `NewParser(runtime.GOOS)`, but it could call `NewParser("")` instead and handle the default case there.

`NewParser()` would check if the passed parameter is the empty string, and default to `runtime.GOOS`.

Reply:
I initially forced the parameter passing to identify where the volume package was called and to try to identify where there could be problems (especially code validating mounts config where we don't yet have access to the image target OS).